src/rust/ide/src/controller/visualization.rs

```diff
@@ -0,0 +1,254 @@
+
+        let paths   = vec![
+            FileSystemObject::new_file(path0.clone()),
+            FileSystemObject::new_file(path1.clone()),
+        ];
+        let file_list_result = language_server::response::FileList{paths};
+        expect_call!(mock_client.file_list(path=path) => Ok(file_list_result));
+
+        let file_content0 = r#"
+            class Vis0 {
+                static inputTypes = ["Float"]
+                onDataReceived(root,data) {}
+                setSize(root,size) {}
+            }
+            return Vis0
+        "#.to_string();
+        let file_content1 = r#"
+            class Vis1 {
+                static inputTypes = ["Float"]
+                onDataReceived(root,data) {}
+                setSize(root,size) {}
+            }
+            return Vis1
+        "#.to_string();
+        let result0 = language_server::response::Read{contents:file_content0.clone()};
+        let result1 = language_server::response::Read{contents:file_content1.clone()};
+        expect_call!(mock_client.read_file(path=path0.clone()) => Ok(result0));
+        expect_call!(mock_client.read_file(path=path1.clone()) => Ok(result1));
+
+        let language_server             = language_server::Connection::new_mock_rc(mock_client);
+        let mut embedded_visualizations = EmbeddedVisualizations::default();
+        let embedded_visualization = Rc::new(class::Handle::new(NativeConstructorClass::new(
+            Signature {
+                name        : "Bubble Visualization (native)".to_string(),
+                input_types : vec!["[[Float,Float,Float]]".to_string().into()],
+            },
+            |scene:&Scene| Ok(Visualization::new(BubbleChart::new(scene)))
+        )));
+        embedded_visualizations.insert("PointCloud".to_string(),embedded_visualization.clone());
+        let vis_controller           = Handle::new(language_server,embedded_visualizations);
+
+        let visualizations = vis_controller.list_visualizations().await;
+        let visualizations = visualizations.expect("Couldn't list visualizations.");
+
+        assert_eq!(visualizations[0],VisualizationIdentifier::Embedded("PointCloud".to_string()));
+        assert_eq!(visualizations[1],VisualizationIdentifier::File(path0));
+        assert_eq!(visualizations[2],VisualizationIdentifier::File(path1));
+
+        let javascript_vis0 = JsSourceClass::from_js_source_raw(&file_content0);
+        let javascript_vis1 = JsSourceClass::from_js_source_raw(&file_content1);
+        let javascript_vis0 = javascript_vis0.expect("Couldn't create visualization class.");
+        let javascript_vis1 = javascript_vis1.expect("Couldn't create visualization class.");
+        let javascript_vis0 = Rc::new(class::Handle::new(javascript_vis0));
+        let javascript_vis1 = Rc::new(class::Handle::new(javascript_vis1));
+
+        let content = vec![embedded_visualization,javascript_vis0,javascript_vis1];
```

Comment:
`content` is extremely generic name here. Perhaps something like `expected_js_classes` ? 
Also I don't think we anywhere else refer to `JsSourceClass` handle as to "content". (it more brings to my mind the "file contents", i.e. before the JsSourceClass instantiation)